sentry_streams/sentry_streams/examples/events.py

```diff
@@ -31,37 +56,96 @@ def build_event(value: str) -> Event:
 
     d: dict[str, Any] = json.loads(value)
 
-    return Event(d["project_id"], d["latency"], d["timestamp"])
+    return Event(d["project_id"], d["latency"], d["timestamp"], d["tags"], d["value"])
 
 
-def build_alert_json(alert: AlertData) -> str:
+def build_alert_json(alert: Union[p95AlertData, CountAlertData]) -> str:
 
     d = alert.to_dict()
 
     return json.dumps(d)
 
 
-class AlertsBuffer(Accumulator[Event, AlertData]):
+class AlertsBuffer(Accumulator[AlertingEvent, Union[p95AlertData, CountAlertData]]):
 
     def __init__(self) -> None:
         self.latencies: list[int] = []
         self.count = 0
+        self.alert_type: str
+        self.alert_id: int
+
+    def add(self, value: AlertingEvent) -> Self:
+        if value.alert_type == "count":
+            self.count += 1
+            self.alert_type = value.alert_type
+            self.alert_id = value.alert_id
 
-    def add(self, value: Event) -> Self:
-        self.latencies.append(value.latency)
-        self.count += 1
+        if value.alert_type == "p95":
+            self.latencies.append(value.latency)
+            self.alert_type = value.alert_type
+            self.alert_id = value.alert_id
 
         return self
 
-    def get_value(self) -> AlertData:
+    def get_value(self) -> Union[p95AlertData, CountAlertData]:
         # A fake p95 calculation, to serve as an example
-        fake_p95 = max(self.latencies)
+        if self.alert_type == "count":
+            return CountAlertData(alert_id=self.alert_id, event_count=self.count)
 
-        return AlertData(p95_latency=fake_p95, event_count=self.count)
+        else:
+            return p95AlertData(alert_id=self.alert_id, p95_latency=max(self.latencies))
 
     def merge(self, other: Self) -> Self:
         # TODO: Use DataSketches
         self.latencies = self.latencies + other.latencies
         self.count = self.count + other.count
 
         return self
+
+
+REGISTERED_ALERTS = {
+    4: {"type": "count", "threshold": 4},
+    5: {"type": "count", "threshold": 2},
+    6: {"type": "p95", "threshold": 4},
+}
+REGISTERED_PROJECT_ALERTS = {2: {"tag_a": 4, "tag_b": 6}, 1: 6}
+
+
+def materialize_alerts(event: Event) -> Generator[AlertingEvent, None, None]:
+
+    project_id = event.project_id
+    alerts_for_project = REGISTERED_PROJECT_ALERTS[project_id]
+
+    if isinstance(alerts_for_project, dict):
+        tags = event.tags
+        for tag in tags:
+            alert_id = alerts_for_project[tag]
+            alert_rule = REGISTERED_ALERTS[alert_id]
+            alert_type = alert_rule["type"]
+            assert isinstance(alert_type, str)
+
+            alerting_event = AlertingEvent(
```

Comment:
I'd call this DataPoint or TimeSeriesDataPoint. 
AlertEvent makes me think about an event emitted when an alert fires.